ospfd/ospf_zebra.c

```diff
@@ -983,21 +1060,22 @@ static int ospf_zebra_read_route(int command, struct zclient *zclient,
 		 * source
 		 * types.
 		 */
-		for (i = 0; i < ZEBRA_ROUTE_MAX; i++)
-			if (i != api.type)
+		for (i = 0; i <= ZEBRA_ROUTE_MAX; i++)
+			if (i != rt_type)
 				ospf_external_info_delete(ospf, i, api.instance,
 							  p);
 
-		ei = ospf_external_info_add(ospf, api.type, api.instance, p,
+		ei = ospf_external_info_add(ospf, rt_type, api.instance, p,
 					    ifindex, nexthop, api.tag);
 		if (ei == NULL) {
 			/* Nothing has changed, so nothing to do; return */
 			return 0;
 		}
-		if (ospf->router_id.s_addr == 0)
+		if (ospf->router_id.s_addr == 0) {
 			/* Set flags to generate AS-external-LSA originate event
 			   for each redistributed protocols later. */
-			ospf->external_origin |= (1 << api.type);
+			ospf->external_origin |= (1 << rt_type);
+			}
 		else {
```

Comment:
if you add bracket around if, please correct style as the bracket must be place at on the same line as the 'else' statement i.e. } else {

Reply:
I have removed the braces for if block since it is a single statement in the if block.